### ISSUE TITLE
properly support progress reporting

### ISSUE THREAD

**Issue by bfrk** [rudymatela/test-framework-leancheck#1]
It would be fantastic if you could improve this module so that one gets progress reports during test runs. At the moment the behavior is such that nothing is printed at all until all leancheck test have been run. This doesn't have to be complicated, see Test.Framework.Providers.SmallCheck. I would do this myself (and issue a pull request) except I think it requires a bit of support from leancheck itself, i.e. some function similar to smallCheckWithHook.

**Comment by rudymatela** [comment]
Makes sense.  I'll try to work on adding this at some point (can't promise it'll be soon).

I think an alternative `checkWithHook` for LeanCheck could live inside `test-framework-leancheck` code and be just based on `Listable.results`.
